types/zfs_fs.nix

```diff
@@ -7,15 +7,11 @@
       description = "Name of the dataset";
     };
     type = lib.mkOption {
-      type = lib.types.enum [ "zfs_dataset" ];
-      default = "zfs_dataset";
+      type = lib.types.enum [ "zfs_fs" ];
+      default = "zfs_fs";
       internal = true;
       description = "Type";
     };
-    zfs_type = lib.mkOption {
-      type = lib.types.enum [ "filesystem" "volume" ];
-      description = "The type of the dataset";
-    };
```

Comment:
That might be useful:

```
  imports = [
    (mkRenamedOptionModule ["zfs_type" ] ["type" ])
  ];
```

Reply:
Than you would also avoid situations like https://github.com/nix-community/disko/issues/197

Reply:
ah, that won't be useful, since the correct type cannot be found. but you get this error right now already:
```
error: No type option set in:
       {
         options = {
           encryption = "aes-256-gcm";
           keyformat = "passphrase";
           keylocation = "file:///tmp/secret.key";
           mountpoint = "none";
         };
         size = "20M";
         zfs_type = "filesystem";
       }
(use '--show-trace' to show detailed location information)
```